File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression/simple_liner_regression.py
# -*- coding: utf-8 -*-
"""
Created on Fri Oct 19 16:29:38 2018

@author: yul04
"""

# import librarys
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# read csv to dataset
dataset = pd.read_csv('Salary_Data.csv')
# iloc, : means all, first parameyter is row, second is column
# X contail all values except last column
X = dataset.iloc[:,:-1].values
# Y contain last column
Y = dataset.iloc[:,1].values

# Split the dataset into the training set and test set
from sklearn.model_selection import train_test_split
# first is independent var, second is dependent var, third is test_size(20% of whole data set), train set is remain 80%, random_state is use for split sampling 
# relation between X and Y base on X -> Y
# test if X-test and Y_test follow predication
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 1/3, random_state = 0)

# make simple linear regression model base on train set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,Y_train)

# predict test set result
Y_pred = regressor.predict(X_test)

#visualising the train set result
plt.scatter(X_train,Y_train,color = 'red')
plt.plot(X_train,regressor.predict(X_train),color = 'blue')
plt.title('Salary Vs. Expreience (Training Set)')
plt.xlabel('Year of Experience')
plt.ylabel('Salary')
plt.show()

#visualising the train set result
plt.scatter(X_test,Y_test,color = 'red')
plt.plot(X_test,regressor.predict(X_test),color = 'blue')
plt.title('Salary Vs. Expreience (Test Set)')
plt.xlabel('Year of Experience')
plt.ylabel('Salary')
plt.show()


# Feature Scaling
# because the data distribution on different dimension might have very large difference in range, so we neeed to scale data
'''
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.fit_transform(X_test)
'''

File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression/simple_liner_regression.R
# read csv
dataset = read.csv('Salary_Data.csv')


# install.packages('caTools')
library(caTools)
# seed value to split
set.seed(123)
# splitRation is ration for training set, first argument is the dependent var(Y)
# return a list contain True/False means split, so for 10 data, 8 true and 2 false
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# fit linear regression to dataset
# lm used to fit linear models
regressor = lm(formula = Salary ~ YearsExperience,
               training_set)

#predit the test set results
y_pred = predict(regressor,newdata = test_set)


#visual train set result
#install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor,newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary Vs. YearsExperience (Train set)') +
  xlab('Years of Experience') +
  ylab('Salary')

#visual test set result
library(ggplot2)
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = test_set$YearsExperience, y = predict(regressor,newdata = test_set)),
            colour = 'blue') +
  ggtitle('Salary Vs. YearsExperience (Test set)') +
  xlab('Years of Experience') +
  ylab('Salary')
